monkey/infection_monkey/network_scanning/scan_target_generator.py

```diff
@@ -52,14 +56,14 @@ def _remove_redundant_targets(targets: List[NetworkAddress]) -> List[NetworkAddr
 def _range_to_addresses(range_obj: NetworkRange) -> List[NetworkAddress]:
     addresses = []
     for address in range_obj:
-        if hasattr(range_obj, "domain_name"):
+        if isinstance(range_obj, HasDomain):
```

Comment:
I reverted this change. I don't like it because it's not straight-forward. We use a type to check for an attribute instead of just checking for an attribute. Mypy complains, but there are two options, either we 1. Create a protocol with optional domain name (but what's the point?) 2. ignore this error. I went for the latter